hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/downloader/PendingDownload.java

```diff
@@ -37,18 +39,21 @@
 @Value
 class PendingDownload {
     private final CompletableFuture<ResponseBytes<GetObjectResponse>> future;
+    private final StreamFilename streamFilename;
     private final Stopwatch stopwatch;
-    private final String s3key; // Source S3 key
+    private final String s3key;
 
     @NonFinal
     private boolean alreadyWaited = false; // has waitForCompletion been called
 
     @NonFinal
     private boolean downloadSuccessful = false;
 
-    PendingDownload(CompletableFuture<ResponseBytes<GetObjectResponse>> future, String s3key) {
+    PendingDownload(CompletableFuture<ResponseBytes<GetObjectResponse>> future, StreamFilename streamFilename,
+            String s3key) {
         this.future = future;
         stopwatch = Stopwatch.createStarted();
+        this.streamFilename = streamFilename;
```

Comment:
If this is only stored to allow for the getFilename() later on then maybe we should consider a minimalistic approach and store the strong rather. Set 
```suggestion
        private final String streamFilename;
        ...
        this.streamFilename = streamFilename.getFilename();
```

Reply:
`getStreamFilename` is called in `parseSignatureFile`

Reply:
Sounds good